test/producePublicFakeFactors.py

```diff
@@ -283,496 +387,59 @@
-    }
-)
-comb_tt_corr_up = replace_nodes(
-    comb, 
-    {'ff_tt':
-     Node(
-         name='ff_tt_corr_up',
-         formula='(1.+{sys_tt})*{ff_tt}',
-         leaves=[
-             Leaf(
-                 name='sys_tt',
-                 file='{HOME}/public/Htautau/FakeRate/{VERSION}/pieces/sys_smooth_tt_nonclosure_mvis.root'.format(HOME=home,VERSION=version),
-                 object='sys_smoothed',
-                 vars=['mvis']
-             ),
-             comb.find('ff_tt')
-         ]
-     )
-    }
-)
-comb_tt_corr_down = replace_nodes(
-    comb, 
-    {'ff_tt':
-     Node(
-         name='ff_tt_corr_down',
-         formula='max(0.,1.-{sys_tt})*{ff_tt}',
-         leaves=[
-             comb_tt_corr_up.find('sys_tt'),
-             comb.find('ff_tt')
-         ]
-     )
-    }
-)
-comb_tt_stat_up = replace_nodes(
-    comb, 
-    {'ff_tt':
-     Node(
-         name='ff_tt_stat_up',
-         formula='(1.+{stat_tt})*{ff_tt}',
-         leaves=[
-             Leaf(
-                 name='stat_tt',
-                 file='{HOME}/public/Htautau/FakeRate/{VERSION}/pieces/toyerr_smooth_tt_mvis.root'.format(HOME=home,VERSION=version),
-                 object='hh_t_mvis_smoothed',
-                 vars=['mvis']
-             ),
-             comb.find('ff_tt')
-         ]
-     )
-    }
-)
-comb_tt_stat_down = replace_nodes(
-    comb, 
-    {'ff_tt':
-     Node(
-         name='ff_tt_stat_down',
-         formula='max(0.,1.-{stat_tt})*{ff_tt}',
-         leaves=[
-             comb_tt_stat_up.find('stat_tt'),
-             comb.find('ff_tt')
-         ]
-     )
-    }
-)
-
-
-## Fill fake factors
-fill(ff_qcd_ss, qcd_ss)
-fill(ff_qcd_ss, qcd_ss_up,   sys='ff_qcd_up')
-fill(ff_qcd_ss, qcd_ss_down, sys='ff_qcd_down')
 fill(ff_qcd_os, qcd_os)
-fill(ff_qcd_os, qcd_os_up,   sys='ff_qcd_up')
-fill(ff_qcd_os, qcd_os_down, sys='ff_qcd_down')
+fill(ff_qcd_os, qcd_os_up,   sys='ff_qcd_syst_up')
+fill(ff_qcd_os, qcd_os_down, sys='ff_qcd_syst_down')
+fill(ff_qcd_os, qcd_os_up_stat,   sys='ff_qcd_stat_up')
+fill(ff_qcd_os, qcd_os_down_stat, sys='ff_qcd_stat_down')
+fill(ff_qcd_os, frac_qcd, sys='frac_qcd')
 fill(ff_w     , w)
-fill(ff_w, w_up,   sys='ff_w_up')
-fill(ff_w, w_down, sys='ff_w_down')
+fill(ff_w, frac_w, sys='frac_w')
+fill(ff_w, frac_dy, sys='frac_dy')
+fill(ff_w, w_up,   sys='ff_w_syst_up')
+fill(ff_w, w_down, sys='ff_w_syst_down')
+fill(ff_w, w_up_stat,   sys='ff_w_stat_up')
+fill(ff_w, w_down_stat, sys='ff_w_stat_down')
 fill(ff_tt    , tt)
-fill(ff_tt, tt_corr_up,   sys='ff_tt_corr_up')
-fill(ff_tt, tt_corr_down, sys='ff_tt_corr_down')
-fill(ff_tt, tt_stat_up,   sys='ff_tt_stat_up')
-fill(ff_tt, tt_stat_down, sys='ff_tt_stat_down')
+fill(ff_tt, tt_up,   sys='ff_tt_syst_up')
+fill(ff_tt, tt_down, sys='ff_tt_syst_down')
+fill(ff_tt, tt_up_stat,   sys='ff_tt_stat_up')
+fill(ff_tt, tt_down_stat, sys='ff_tt_stat_down')
+fill(ff_tt, frac_tt, sys='frac_tt')
 fill(ff_comb  , comb)
-fill(ff_comb, comb_qcd_up,   sys='ff_qcd_up')
```

Comment:
I don't think it will work if you don't register the systematics in the combined fake factor